peps/pep-0776.rst

```diff
@@ -0,0 +1,794 @@
+caught, it displays the JavaScript exception and calls ``_Py_DumpTraceback()``.
+It then exits with code 1. We will stick with this approach until we add either
+a JavaScript API or foreign function interface, which is out of scope for this PEP.
+
+Specification
+=============
+
+Scope of Work
+-------------
+
+Adding Emscripten as a Tier 3 platform only requires adding support for
+compiling an Emscripten-compatible build from the unpatched CPython source code.
+It does not necessarily require there to be any officially distributed
+Emscripten artifacts on python.org, although these could be added in the future.
+In the short term, they will continue to be distributed downstream with Pyodide.
+
+Emscripten will be built using the same configure and Makefile system as other
+POSIX platforms, and must therefore be built on a POSIX platform. Both Linux and
+macOS will be supported.
+
+A Python CLI entrypoint will be provided, which among other things can be used
+to execute the test suite.
+
+Linkage
+-------
+
+It is only supported to statically link the Python interpreter. We use `EM_JS
+<https://emscripten.org/docs/api_reference/emscripten.h.html#c.EM_JS>`__
+functions in the interpreter for various purposes. It is possible to dynamically
+link object files that include ``EM_JS`` functions, but their behavior deviates
+significantly from their behavior in static builds. For this reason, it would
+require special work to support. If a use case for dynamically linking the
+interpreter in Emscripten emerges, we can evaluate how much effort would be
+required to support it.
+
+Standard Library
+----------------
+
+Unsupported Modules
+~~~~~~~~~~~~~~~~~~~
+
+See https://pyodide.org/en/stable/usage/wasm-constraints.html#removed-modules.
+
+Removed Modules
+^^^^^^^^^^^^^^^
+
+The following modules are removed from the standard library to reduce download
+size and since they currently wouldn't work in the WebAssembly VM.
+
+- curses
+- dbm
+- ensurepip
+- fcntl
+- grp
+- idlelib
+- msvcrt
+- pwd
+- resource
+- syslog
+- termios
+- tkinter
+- turtle
+- turtledemo
+- venv
+- winreg
+- winsound
+
+Included but not Working Modules
+^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
+
+The following modules can be imported, but are not functional:
+
+- multiprocessing
+- threading
+- sockets
+
+as well as any functionality that requires these.
+
+The following are present but cannot be imported due to a dependency on the
+termios package which has been removed:
+
+- pty
+- tty
+
+
+Platform Identification
+~~~~~~~~~~~~~~~~~~~~~~~
+
+``sys.platform`` will return ``"emscripten"``. Although Emscripten attempts to
+be compatible with Linux, the differences are significant enough that a distinct
+name is justified. This is consistent with the return value from ``os.uname()``.
+
+There is also ``sys._emscripten_info`` which includes the Emscripten version and
+the runtime (either ``navigator.userAgent`` in a browser or ``"Node js" +
+process.version`` in Node.js).
```

Comment:
The discussion around PEP 730 and PEP 738 concluded that `platform` was a better place for a public API for this sort of data (recognising that there may be a need to have it embedded in `sys` as a compiled property). There's probably space for an `emscripten_ver()` analog to `mac_ver()`, `ios_ver()` and `android_ver()`.